packages/react-utilities/src/compose/types.ts

```diff
@@ -101,7 +102,7 @@ export type ComponentProps<
   {
     [Key in keyof Shorthands]?: ShorthandProps<NonNullable<Shorthands[Key]>>;
   },
-  Primary
+  Primary & 'root' // Omit the 'root' prop only if it is the Primary slot (which is the default case)
```

Comment:
Is this addition necessary?

Reply:
Yes. This is necessary to ensure that the `root` slot prop exists for components whose primary slot is something other than root. And the `root` slot is necessary so that users can set properties on the root element, since they'd go to the primary slot otherwise.

Reply:
Actually to clarify, this `& 'root'` is also making sure that the primary slot exists on those components, and the primary slot is necessary for setting style and className on the primary slot. In short this is doing:
* If Primary is 'root', then omit the `root` slot
* Otherwise, include both Primary and `root` slots, since they are both necessary

Reply:
Updated the comment to clarify what this is doing